rotkehlchen/db/snapshots.py

```diff
@@ -202,17 +202,20 @@ def _export(
 
     def import_snapshot(
             self,
+            write_cursor: 'DBCursor',
             processed_balances_list: List[DBAssetBalance],
             processed_location_data_list: List[LocationData],
     ) -> Tuple[bool, str]:
         """Import the validated snapshot data to the database."""
-        with self.db.user_write() as cursor:
-            self.add_nft_asset_ids(cursor, [entry.asset.identifier for entry in processed_balances_list])  # noqa: E501
-            try:
-                self.db.add_multiple_balances(cursor, processed_balances_list)
-                self.db.add_multiple_location_data(cursor, processed_location_data_list)
-            except InputError as err:
-                return False, str(err)
+        self.add_nft_asset_ids(
+            write_cursor=write_cursor,
+            entries=[entry.asset.identifier for entry in processed_balances_list],
+        )
+        try:
+            self.db.add_multiple_balances(write_cursor, processed_balances_list)
+            self.db.add_multiple_location_data(write_cursor, processed_location_data_list)
+        except InputError as err:
```

Comment:
Hey @prettyirrelevant are you reading my reviews? https://github.com/rotki/rotki/pull/4526/files#r913142149

This is wrong. I already said that before.

If anything fails inside there you catch it here and do not let it propagate up to the context manager so nothign is rolled back.

Can you please read the code, understand how it works and ask me if something is not clear instead of pushing stuff and resolving my comments without actually solving anything?

Reply:
Rollback part here: https://github.com/rotki/rotki/pull/4532